e2e/app/element/primereact-input-number.ts

```diff
@@ -0,0 +1,42 @@
+import BaseElement from './base-element';
+import {ElementHandle, Page} from 'puppeteer';
+
+export default class PrimereactInputNumber extends BaseElement {
+  constructor(page: Page, xpath?: string) {
+    super(page, xpath);
+  }
+
+  async setValue(value: number): Promise<void> {
+    const input = await this.getInputElementHandle();
+    // primereact InputNumber thoughtfully ignores any input that would cause the input to go outside
+    // its specified mins and maxes and is very finicky around 1) copy/paste and 2) manual manipulation
+    // of its value prop so instead we must do shenanigans!
+    await input.click();
+    while (await this.getInputValue() !== value) {
+      const curr = await this.getInputValue();
+      if (curr > value) {
+        await this.page.keyboard.press('ArrowDown');
+      }
+      else if (curr < value) {
+        await this.page.keyboard.press('ArrowUp');
+      }
+    }
+  }
+
+  async getInputValue(): Promise<number> {
+    const input = await this.getInputBaseElement();
+    return parseInt(await input.getProperty('value'));
```

Comment:
how about `await getPropValue<number>(input, 'value');`? (I didn't test it out).

Reply:
tbh we're thinking about not using this sort of input anymore cause it has serious UX problems so I don't want to spend any more time messing with it